cpp/arcticdb/util/query_stats.cpp

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+*/
+
+#include <arcticdb/util/query_stats.hpp>
+#include <arcticdb/async/task_scheduler.hpp>
+#include <arcticdb/util/preconditions.hpp>
+#include <arcticdb/log/log.hpp>
+
+namespace arcticdb::util::query_stats {
+
+
+OpStats::OpStats() : 
+    logical_key_counts_{0},
+    result_count_(0),
+    total_time_ms_(0),
+    count_(0) {
+
+}
+
+void OpStats::reset_stats() {
+    for (auto& logical_key_count : logical_key_counts_) {
+        logical_key_count = 0;
+    }
+    result_count_ = 0;
+    total_time_ms_ = 0;
+    count_ = 0;
+}
+
+
+Stats::Stats() : 
+    keys_stats_{} {
+
+}
+
+void Stats::reset_stats() {
+    for (auto& op_stats : keys_stats_) {
+        for (auto& op_stat : op_stats) {
+            op_stat.reset_stats();
+        }
+    }
+}
+
+QueryStats QueryStats::instance_;
+
+QueryStats& QueryStats::instance() {
+    return instance_;
+}
+
+void QueryStats::reset_stats() {
+    check(!async::TaskScheduler::instance()->tasks_pending(), "Folly tasks are still running");
```

Comment:
Doesn't this mean we'll blow up if there are multiple Python threads running? I don't think we should let the stats measurements interrupt a program